core/src/main/java/com/venia/core/models/commerce/services/CommerceComponentModelFinder.java

```diff
@@ -0,0 +1,121 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.venia.core.models.commerce.services;
+
+import com.adobe.cq.commerce.core.components.models.product.Product;
+import com.adobe.cq.commerce.core.components.models.productlist.ProductList;
+import com.drew.lang.annotations.Nullable;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.factory.ModelFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * This service allows to traverse a {@link Resource} tree looking for a {@link Resource} of a set of particular resource types and if
+ * found adapting them to given adapter type. This helps for example finding the product component on the page and return the Product model
+ * from it.
+ */
+@Component(
+        service = com.venia.core.models.commerce.services.CommerceComponentModelFinder.class)
+public class CommerceComponentModelFinder {

```

Comment:
I think there are methods in this class which were copied over by accident but they are not used in the actual production code and you still had to provide test coverage for them. I think we should try to keep the example code clean and simple as much as possible to avoid confusing users with needless accidental complexity.
Could you please remove the unneeded methods here. There is also a completely unused method.

Reply:
Done